kubel-evil.el

```diff
@@ -53,6 +53,13 @@
 (require 'kubel)
 
 ;;; Code:
+(define-minor-mode kubel-evil-mode
+  "Brings evil keybindings to kubel"
+  :lighter " kubel-evil"
+  :keymap (make-sparse-keymap)
```

Comment:
I believe instead of creating a new keymap here, you could:
```elisp
(defvar kubel-evil-mode-map
    (let ((keymap (make-sparse-keymap)))
        ;; modify the keymap however you like
        keymap)
```
And then:
```suggestion
  :keymap kubel-evil-mode-map
```

Reply:
What advantages does the `let` block have to let's say this?:
```elisp
(defvar kubel-evil-mode-map (make-sparse-keymap))
```

Reply:
I was wrong about this -- there's an idiom where you defvar and define the keymap in a single block, which would've looked like this:
```elisp
;; THIS IS WRONG
(defvar kubel-evil-mode-map
  (let ((keymap (make-sparse-keymap)))
    (evil-define-key 'motion shx-mode-map
      (kbd "RET") #'kubel-get-pod-details
      ;;...
      (kbd "a") #'kubel-jab-deployment)
    keymap))
```
But I think this doesn't work with evil because of how it stores keymaps (q.v. `evil-minor-mode-keymaps-alist`).  So there's no advantage in this case to the let block, but the `defvar` will give the keymap a name.

...which I think you need to use down [here](https://github.com/abrochard/kubel/pull/8/files#diff-8058de983aa58ab124e879594ca5315fL73) so that you're modifying the minor mode's map instead of `kubel-mode-map`.

Let me know if I'm making any sense. :)

Reply:
Yes it makes sense. Before I was overriding the original keymap of kubel. But now that I fill the keybindings into my own keymap it didn't work first. That's why I had to add a [defgroup](https://github.com/abrochard/kubel/pull/8/files#diff-8058de983aa58ab124e879594ca5315fR55), which integrates my group into the one of kubel. This is how [evil-org-mode](https://github.com/Somelauw/evil-org-mode/blob/master/evil-org.el#L46) does it as well. I think now it should be like intended right?

Reply:
That looks right!  Thanks again for accommodating and figuring this out -- I'll be referring back to this package in the future I'm sure.

Reply:
thanks and ty for your help on getting this done :+1: